osu.Game/Graphics/UserInterface/OsuPasswordTextBox.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu-framework/master/LICENCE
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using OpenTK;
+using OpenTK.Graphics;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class OsuPasswordTextBox : OsuTextBox
+    {
+        protected override Drawable GetDrawableCharacter(char c) => new PasswordMaskChar(CalculatedTextSize);
+
+        public class PasswordMaskChar : Container
+        {
+            private CircularContainer circle;
+
+            public PasswordMaskChar(float size)
+            {
+                Size = new Vector2(size / 2, size);
+                Children = new[]
+                {
+                    circle = new CircularContainer

```

Comment:
what would you recommend? since the size is relative, the cornerradius is unknown at the point of instantiating so i'd have to do the same update override locally..